src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -326,30 +326,28 @@ public OrganisationsDetailResponse findByOrganisationStatus(OrganisationStatus s
     @Override
     @Transactional
     public DeleteOrganisationResponse deleteOrganisation(Organisation organisation) {
-        DeleteOrganisationResponse deleteOrganisationResponse;
+        DeleteOrganisationResponse deleteOrganisationResponse = new DeleteOrganisationResponse();
         if (OrganisationStatus.PENDING == organisation.getStatus()) {
-            deleteOrganisationResponse = deleteOrganisationEntity(organisation);
+            deleteOrganisationResponse = deleteOrganisationEntity(organisation, deleteOrganisationResponse);
         } else if (OrganisationStatus.ACTIVE == organisation.getStatus()) {
-            deleteOrganisationResponse = deleteUserProfile(organisation);
+            deleteOrganisationResponse = deleteUserProfile(organisation, deleteOrganisationResponse);
             deleteOrganisationResponse = deleteOrganisationResponse.getStatusCode() == ProfessionalApiConstants.STATUS_CODE_204
-                    ? deleteOrganisationEntity(organisation) : deleteOrganisationResponse;
+                    ? deleteOrganisationEntity(organisation, deleteOrganisationResponse) : deleteOrganisationResponse;
         } else  {
             throw new EmptyResultDataAccessException(ProfessionalApiConstants.ONE);
         }
         return deleteOrganisationResponse;
```

Comment:
you can also do this with switch instead of if-else

Reply:
as suggested updated to switch condition